### PR TITLE
feat: adding back bounds

### PR DESCRIPTION
**Pull request by Andarius**
## Description

- Adding  `bounds` to `<MapboxGL.Camera />`.
- Fix for padding with `bounds`.
- Adding None to `animationMode`.
- Fix zoom/scroll/rotate/logo enabled/disabled

## Checklist

- [ ] I have tested this on a device/simulator for each compatible OS
- [ ] I updated the documentation with running `yarn generate` in the root folder
- [ ] I mentioned this change in `CHANGELOG.md`
- [X] I updated the typings files (`index.d.ts`)
- [ ] I added/ updated a sample (`/example`)

### CODE REVIEW

**Review by felixgourdeau — APPROVED** [review]

**Comment by Andarius** [comment]
I'm doing some testing, and it appears that when rendering more than 2 maps, the zoom cannot be computed
`E/Mapbox: [maps-core]: {obitrain.obiapp}[General]: Unable to calculate appropriate zoom level for bounds. Vertical or horizontal padding is greater than map's height or width.` 
Not sure why, maybe related to https://github.com/rnmapbox/maps/pull/1057 ?

File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/mapview/RCTMGLMapViewManager.kt [hunk 1]
```diff
@@ -94,23 +97,25 @@ open class RCTMGLMapViewManager(context: ReactApplicationContext?) :
...
-        //mapView.setReactPitchEnabled(pitchEnabled);
+    fun setPitchEnabled(mapView: RCTMGLMapView, pitchEnabled: Boolean) {
+        mapView.gestures.pitchEnabled = pitchEnabled
     }
 
     @ReactProp(name = "rotateEnabled")
-    fun setRotateEnabled(mapView: RCTMGLMapView?, rotateEnabled: Boolean) {
-        //mapView.setReactRotateEnabled(rotateEnabled);
+    fun setRotateEnabled(mapView: RCTMGLMapView, rotateEnabled: Boolean) {
+        mapView.gestures.rotateEnabled
```
Comment (felixgourdeau) [inline android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/mapview/RCTMGLMapViewManager.kt:118]: `mapView.gestures.rotateEnabled = rotateEnabled`

File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/camera/CameraStop.java [hunk 2]
```diff
@@ -180,10 +166,10 @@ public static CameraStop fromReadableMap(Context context, @NonNull ReadableMap r
         }
 
         if (readableMap.hasKey("bounds")) {
-            int paddingTop = getBoundsPaddingByKey(readableMap, "boundsPaddingTop");
```
Comment (mfazekas) [inline android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/camera/CameraStop.java:183]: Are you sure about the `boundsPaddingTop` => `paddingTop` rename?! I think on JS part we still use `boundsPaddingTop` as I looked.

Comment (mfazekas) [reply android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/camera/CameraStop.java:183]: Ok it seems that here we still use the old names.
https://github.com/rnmapbox/maps/blob/1ae89df048841327c871553f76a8d21181ed13c3/javascript/components/MapView.js#L632-L635
But other native code uses `paddingTop` not `boundsPaddingTop` so this change seems to be ok.

This was likely changed in `https://github.com/rnmapbox/maps/pull/1538`. I mean boundsPaddingTop to just paddingTop.

Comment (Andarius) [reply android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/camera/CameraStop.java:183]: Should I mark this as resolved or do you want me to make some changes?

File: javascript/components/Camera.js [hunk 3]
```diff
@@ -616,6 +616,7 @@ class Camera extends React.Component {
       case Camera.Mode.Linear:
         return MapboxGL.CameraModes.Linear;
       case Camera.Mode.None:
+        return MapboxGL.CameraModes.None;
       case Camera.Mode.Move:
         return MapboxGL.CameraModes.Move;
```
Comment (mfazekas) [inline javascript/components/Camera.js:621]: `None` is same as `Move`, but it's a bit of a mess as most implementation exports `MapboxGL.CameraModes.None`, but iOS on v10 only export `Move`

Comment (Andarius) [reply javascript/components/Camera.js:621]: How do you disable the animation on iOS?

Comment (mfazekas) [reply javascript/components/Camera.js:621]: `Move` just jumps without animation. 

https://github.com/rnmapbox/maps/pull/1838#issuecomment-1100638219

**Comment by felixgourdeau** [comment]
@mfazekas @Andarius should we merge this PR for now, and fix the issue with more than 2 maps?

**Comment by mfazekas** [comment]
> @mfazekas @Andarius should we merge this PR for now, and fix later the issue with more than 2 maps?

Yes sorry, let's merge this. Please update code so it can be merged.

**Review by felixgourdeau — APPROVED** [review]

File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/mapview/RCTMGLMapViewManager.kt [hunk 4]
```diff
@@ -131,8 +136,8 @@ open class RCTMGLMapViewManager(context: ReactApplicationContext?) :
     }
 
     @ReactProp(name = "compassEnabled")
-    fun setCompassEnabled(mapView: RCTMGLMapView?, compassEnabled: Boolean) {
-        mapView!!.setReactCompassEnabled(compassEnabled!!);
```
Comment (mfazekas) [inline android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/mapview/RCTMGLMapViewManager.kt:135]: Is this intentional?! 


setReactCompassEnabled sound like a better implementation

https://github.com/rnmapbox/maps/blob/main/android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/mapview/RCTMGLMapView.kt#L855

File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/mapview/RCTMGLMapViewManager.kt [hunk 5]
```diff
@@ -121,8 +126,8 @@ open class RCTMGLMapViewManager(context: ReactApplicationContext?) :
     }
 
     @ReactProp(name = "logoEnabled")
-    fun setLogoEnabled(mapView: RCTMGLMapView?, logoEnabled: Boolean?) {
-        mapView!!.setReactLogoEnabled(logoEnabled);
```
Comment (mfazekas) [inline android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/mapview/RCTMGLMapViewManager.kt:125]: What's wrong with `setReactLogoEnabled` ?!

**Review by mfazekas — CHANGES_REQUESTED** [review]
I think logoEnabled, and compassEnabled changes needs revert

File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/mapview/RCTMGLMapViewManager.kt (hunk 4 above)
Comment (Andarius) [reply android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/mapview/RCTMGLMapViewManager.kt:135]: You are right, my bad. I'm reverting

Comment (Andarius) [reply android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/mapview/RCTMGLMapViewManager.kt:135]: Just for information, what's the point of `compassEnabled!!` and not just `compassEnabled` ?

Comment (mfazekas) [reply android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/mapview/RCTMGLMapViewManager.kt:135]: I guess it's a mistake, feel free to remove `!!`

**Review by mfazekas — APPROVED** [review]

**Pull request merged**